hartshorn-core/src/main/java/org/dockbox/hartshorn/core/domain/Exceptional.java

```diff
@@ -240,7 +253,8 @@ public T orNull() {
      * @return The value, if present, otherwise {@code other}
      * @see Exceptional#orNull()
      */
-    public T or(final T other) {
+    @NonNull
+    public T or(@NonNull final T other) {
```

Comment:
`other` can be nullable, as described in the JavaDoc of this method

Reply:
Wouldn't we have to annotate the return value with `@Nullable` too then?

Reply:
I feel as though it makes more sense to enforce it being `@NonNull` here so that we can keep the result guaranteed to be not null and instead make them use `orNull`

Reply:
It may not always be known whether the value passed into `or` will be null or non-null, though I agree with your motivation for it being `NonNull` as it also forces you to think about what could be returned. `NonNull` is fine here 👍🏼